library/general/src/modules/Report.rb

```diff
@@ -594,15 +598,17 @@ def Error(error_string)
 
     # Store new error text, the text is displayed in a richtext widget - long lines are automatically wrapped
     # @param [String] error_string error text  (it can contain rich text tags)
+    # @param width [Integer] width of popup (@see Popup#LongErrorGeometry)
+    # @param height [Integer] height of popup (@see Popup#LongErrorGeometry)
     # @return [void]
-    def LongError(error_string)
+    def LongError(error_string, width: 60, height: 10)
```

Comment:
Hm, optional parameters....will that work via the YaST component system? The problem is that the `Report` module might be used from Perl code (yast2-users)....

Reply:
For the record: according to @jreidinger it should work across languages.

Reply:
For the record let me describe it how it works.

Ruby bindings uses `publish` to get API with types. Limitation is that publish support only single signature. So in this case one, two or three params can be used, but only one have to picked. When called this method from e.g. perl, it will use ruby calls, so fulfill all missing params with its default value. But keyword arguments is not supported by ruby bindings and component system, so probably some hash workaround have to be used. In this case publish has only string param and no keyword ones, so it should work without issues.